pkg/provisioner/ironic/ironic.go

```diff
@@ -1221,6 +1221,59 @@ func (p *ironicProvisioner) Prepare(data provisioner.PrepareData, unprepared boo
 	return
 }
 
+func (p *ironicProvisioner) getConfigDrive(data provisioner.ProvisionData) (result nodes.ConfigDrive, err error) {
+	// Retrieve instance specific user data (cloud-init, ignition, etc).
+	userData, err := data.HostConfig.UserData()
+	if err != nil {
+		return result, errors.Wrap(err, "could not retrieve user data")
+	}
+	if userData != "" {
+		result.UserData = userData
+	}
+
+	// Retrieve OpenStack network_data. Default value is empty.
+	networkDataRaw, err := data.HostConfig.NetworkData()
+	if err != nil {
+		return result, errors.Wrap(err, "could not retrieve network data")
+	}
+	if networkDataRaw != "" {
+		var networkData map[string]interface{}
+		if err = yaml.Unmarshal([]byte(networkDataRaw), &networkData); err != nil {
+			return result, errors.Wrap(err, "failed to unmarshal network_data.json from secret")
+		}
+		result.NetworkData = networkData
+	}
+
+	// Retrieve meta data with fallback to defaults from provisioner.
+	metaData := map[string]interface{}{
+		"uuid":             string(p.objectMeta.UID),
+		"metal3-namespace": p.objectMeta.Namespace,
+		"metal3-name":      p.objectMeta.Name,
+		"local-hostname":   p.objectMeta.Name,
+		"local_hostname":   p.objectMeta.Name,
+		"name":             p.objectMeta.Name,
+	}
+	metaDataRaw, err := data.HostConfig.MetaData()
+	if err != nil {
+		return result, errors.Wrap(err, "could not retrieve metadata")
+	}
+	if metaDataRaw != "" {
+		if err = yaml.Unmarshal([]byte(metaDataRaw), &metaData); err != nil {
+			return result, errors.Wrap(err, "failed to unmarshal metadata from secret")
+		}
+	}
+
+	// Set metaData if any field is populated by a user.
+	if metaDataRaw != "" || networkDataRaw != "" || userData != "" {
+		result.MetaData = metaData
+		p.log.Info("triggering provisioning with config drive")
+	} else {
+		p.log.Info("triggering provisioning without config drive")
```

Comment:
I'm not sure this is correct, as it changes the existing behavior - above we default `metaData` then also consume any user-provided metaDataRaw, e.g we should always create a configDrive containing at least the default metaData?